src/test/java/io/aiven/kafka/connect/transforms/ConcatFieldsTest.java

```diff
@@ -0,0 +1,271 @@
+        newValueMap.put(FIELD, TEST_VALUE.toString() + "-" + FOO_VALUE + "-" + BAR_VALUE + "-" + AGE_VALUE);
+
+        final SinkRecord originalRecord = record(null, valueMap);
+        final SinkRecord transformedRecord = transformation().apply(originalRecord);
+        assertEquals(setNewValue(originalRecord, null, newValueMap), transformedRecord);
+    }
+
+    @Test
+    void mapWithoutSchemaMissingField() {
+        final HashMap<Object, Object> valueMap = new HashMap<>();
+        valueMap.put(TEST_FIELD, TEST_VALUE);
+        valueMap.put(BAR_FIELD, BAR_VALUE);
+        valueMap.put(AGE_FIELD, AGE_VALUE);
+        final HashMap<Object, Object> newValueMap = new HashMap<>();
+        newValueMap.put(TEST_FIELD, TEST_VALUE);
+        newValueMap.put(BAR_FIELD, BAR_VALUE);
+        newValueMap.put(AGE_FIELD, AGE_VALUE);
+        newValueMap.put(FIELD, TEST_VALUE.toString() + "-*-" + "Baz" + "-" + AGE_VALUE);
+
+        final SinkRecord originalRecord = record(null, valueMap);
+        final SinkRecord transformedRecord = transformation().apply(originalRecord);
+        assertEquals(setNewValue(originalRecord, null, newValueMap), transformedRecord);
+    }
+
+    @Test
+    void mapWithSchema() {
+        final HashMap<Object, Object> valueMap = new HashMap<>();
+        valueMap.put(TEST_FIELD, TEST_VALUE);
+        valueMap.put(FOO_FIELD, FOO_VALUE);
+        valueMap.put(BAR_FIELD, BAR_VALUE);
+        valueMap.put(AGE_FIELD, AGE_VALUE);
+        final HashMap<Object, Object> newValueMap = new HashMap<>();
+        newValueMap.put(TEST_FIELD, TEST_VALUE);
+        newValueMap.put(FOO_FIELD, FOO_VALUE);
+        newValueMap.put(BAR_FIELD, BAR_VALUE);
+        newValueMap.put(AGE_FIELD, AGE_VALUE);
+        newValueMap.put(FIELD, TEST_VALUE.toString() + "-" + FOO_VALUE + "-" + BAR_VALUE + "-" + AGE_VALUE);
+
+        final SinkRecord originalRecord = record(VALUE_SCHEMA, valueMap);
+        final SinkRecord transformedRecord = transformation().apply(originalRecord);
+        assertEquals(setNewValue(originalRecord, NEW_VALUE_SCHEMA, newValueMap).toString(),
+            transformedRecord.toString());
+    }
+
+    @Test
+    void mapWithSchemaMissingField() {
+        final HashMap<Object, Object> valueMap = new HashMap<>();
+        valueMap.put(TEST_FIELD, TEST_VALUE);
+        valueMap.put(BAR_FIELD, BAR_VALUE);
+        valueMap.put(AGE_FIELD, AGE_VALUE);
+        final HashMap<Object, Object> newValueMap = new HashMap<>();
+        newValueMap.put(TEST_FIELD, TEST_VALUE);
+        newValueMap.put(BAR_FIELD, BAR_VALUE);
+        newValueMap.put(AGE_FIELD, AGE_VALUE);
+        newValueMap.put(FIELD, TEST_VALUE.toString() + "-*-" + "Baz" + "-" + AGE_VALUE);
+
+        final SinkRecord originalRecord = record(VALUE_SCHEMA, valueMap);
+        final SinkRecord transformedRecord = transformation().apply(originalRecord);
+        assertEquals(setNewValue(originalRecord, NEW_VALUE_SCHEMA, newValueMap).toString(),
```

Comment:
No need to call `toString()`, `SinkRecords` are comparable (in some other places too).